src/gsi/read_prepbufr.f90

```diff
@@ -1914,6 +1922,26 @@ subroutine read_prepbufr(nread,ndata,nodata,infile,obstype,lunout,twindin,sis,&
 !             Missing Values ==>  Cycling! In this case for howv only.  #ww3 
               if (howvob  .and. owave(1,k) > r0_1_bmiss) cycle LOOP_K_LEVS
 
+!             Over-ride QM=9 and hard-wire errors for land obs and hofx_sfc_file option
+!             Can be deleted once prepbufr processing updated.
+              if ( global_2m_land ) then
+                if (tob .and. qm==9 ) then
+                     pqm(k)=2 ! otherwise, type 183 will be discarded.
+                     qm=2
+                     tqm(k)=2
+                     if (kx==187) obserr(3,k)=2.2
+                     if (kx==181) obserr(3,k)=1.5
+                     if (kx==183) obserr(3,k)=2.6
+                endif
+                if (qob .and. qm == 9 ) then 
+                     qm = 2
+                     ! qob err specified as fraction of qsat, multiplied by 10.
+                     if (kx==187) obserr(2,k)=1.0
+                     if (kx==181) obserr(2,k)=1.0
+                     if (kx==183) obserr(2,k)=1.0
+                endif
+
+              endif
```

Comment:
I was assuming that this section would give me trouble!

I'm not sure what you mean by the "authoritative repository", but as indicated by the comment, we only need this code temporarily to over-ride the current values in the prepbufr files. I'd like to merge this code into develop if I can (a safer, but perhaps uglier, solution would be to comment it out, but keep it). I'm assuming that well before this code becomes operational, we'll have updated the prepbufr processing, and this can be deleted.

I don't yet know what error values we want to use, so it's probably too early to change the prepbufr processing, but if someone wants to point me to the relevant repo I can create an issue there.

Reply:
I'll follow up with Jeff.

Reply:
Jeff doesn't have that tool anymore (in any case, I think it'll be just as easy for me to patch my code if we don't want to add this code to the repo). 

@RussTreadon-NOAA What would you like me to do here? Currently, the prepbufr files have a default quality mark that prevents the screen-level T and q from being assimilated (and no error info for these obs). I've added a block of code to overwrite this, if hofx_2m_sfcfile is true. The only reason I'm hesitating to delete it is that there is currently no way to test the new screen-level assimilation code without this block. Perhaps I could delete the code block, and replace it with a comment recording which values (quality marks, and errors) would need to be over-written?